client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,83 @@
+import { Controller } from '@hotwired/stimulus';
+import { object } from 'prop-types';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ * id="{id}"
+ * data-block
+ * data-controller="w-block"
+ * data-w-block-data-value="{block_json}"
+ * data-w-block-initial-value="{value_json}"
+ * data-w-block-error-value="{error_json}">
+ * </div>
+ */
+
+export class BlockController extends Controller<HTMLElement> {
```

Comment:
There is still a line break after the JSDoc, here's a suggested fix with some more refinements to the example HTML also. We do not need much in this example but need to assume that the Django template syntax is not to be used in the example.

```suggestion
/**
 * Adds the ability to unpack a Telepath object and render it on the controlled element.
 *
 * @example
 * <div
 *  id="some-id"
 *  data-controller="w-block"
 *  data-w-block-data-value='{"some":"json"}'
 * >
 * </div>
 */
export class BlockController extends Controller<HTMLElement> {
```